modules/hal_nxp/usb/usb_dc_mcux.h

```diff
@@ -25,6 +25,43 @@
 #define USB_DEVICE_CONFIG_SELF_POWER (1U)
 #endif
 
+#ifdef CONFIG_USB_DC_NXP_LPCIP3511
+
+#ifdef USBHSD_BASE_ADDRS
+#define USB_DEVICE_CONFIG_LPCIP3511HS (1U)
+#else
+#define USB_DEVICE_CONFIG_LPCIP3511HS (0U)
+#endif
+
+#ifdef USB_BASE_ADDRS
+#define USB_DEVICE_CONFIG_LPCIP3511FS (1U)
+#else
+#define USB_DEVICE_CONFIG_LPCIP3511FS (0U)
+#endif
+
+/* Whether device is self power. 1U supported, 0U not supported */
+#define USB_DEVICE_CONFIG_SELF_POWER (1U)
+
+#endif
+
+/* big/little endian */
+#define SWAP2BYTE_CONST(n) ((((n)&0x00FFU) << 8U) | (((n)&0xFF00U) >> 8U))
+
+#if defined(CONFIG_BIG_ENDIAN)
+#define USB_SHORT_FROM_LITTLE_ENDIAN(n) SWAP2BYTE_CONST(n)
+#else
+#define USB_SHORT_FROM_LITTLE_ENDIAN(n) (n)
+#endif
```

Comment:
Use helper functions in `include/sys/byteorder.h`

Reply:
Thank you, I have made the change.